app/src/main/kotlin/pl/elpassion/project/choose/ProjectChooseController.kt

```diff
@@ -2,22 +2,40 @@ package pl.elpassion.project.choose
 
 import pl.elpassion.project.Project
 import pl.elpassion.project.ProjectRepository
+import rx.Observable
 
-class ProjectChooseController(val view: ProjectChoose.View, val repository: ProjectRepository) {
-    private lateinit var sortedProjectsList: List<Project>
+class ProjectChooseController(private val view: ProjectChoose.View,
+                              repository: ProjectRepository) {
+
+    private val projectsObservable: Observable<Project> =
+            repository.getProjects()
+                    .flatMapIterable { it }
+                    .sorted { project, project2 -> project.name.compareTo(project2.name) }
+                    .replay()
+                    .autoConnect()
 
     fun onCreate() {
-        sortedProjectsList = repository.getPossibleProjects().sortedBy { it.name }
-        view.showPossibleProjects(sortedProjectsList)
+        projectsObservable.subscribeProjects()
     }
 
     fun onProjectClicked(project: Project) {
         view.selectProject(project)
     }
 
-    fun searchQuery(query: String) {
-        view.showFilteredProjects(filterProjectByQuery(query))
+    private fun Observable<Project>.subscribeProjects() {
```

Comment:
Underneath this subscription can be call to api, so on the screen should be shown loader.